src/server/api/AdminPanel.ts

```diff
@@ -0,0 +1,131 @@
+				return;
+			}
+			deleteUser(getMemberDataByCriteria(["user_id"], "username", commandAndArgs[1])["user_id"], commandAndArgs[2] ?? "");
+			res.status(200).send("");
+			logEvents("Command executed: " + command + "\n", "adminCommands");
+			return;
+		case "username":
+			if (commandAndArgs[1] === "get") {
+				if (commandAndArgs.length < 3) {
+					res.status(400).send("Invalid number of arguments, expected 2, got " + (commandAndArgs.length - 1) + ".");
+					return;
+				}
+				res.status(200).send(getMemberDataByCriteria(["username"], "user_id", Number.parseInt(commandAndArgs[2]!))["username"]);
+				logEvents("Command executed: " + command + "\nResult: " + getMemberDataByCriteria(["username"], "user_id", Number.parseInt(commandAndArgs[2]!))["username"] + "\n", "adminCommands");
+			}
+			else if (commandAndArgs[1] === "set") {
+				if (commandAndArgs.length < 4) {
+					res.status(400).send("Invalid number of arguments, expected 3, got " + (commandAndArgs.length - 1) + ".");
+					return;
+				}
+				// TODO add username changing logic
+				res.status(503).send("Changing usernames is not yet supported.");
+			}
+			else {
+				res.status(400).send("Invalid subcommand, expected either get or set, got " + commandAndArgs[1] + ".");
+			}
+			return;
+		case "logout":
+			if (commandAndArgs.length < 2) {
+				res.status(400).send("Invalid number of arguments, expected 1, got " + (commandAndArgs.length - 1) + ".");
+				return;
+			}
+			deleteRefreshTokensOfUser(getMemberDataByCriteria(["user_id"], "username", commandAndArgs[1])["user_id"]);
+			res.status(200).send("");
+			logEvents("Command executed: " + command + "\n", "adminCommands");
+			return;
+		case "verify":
+			return;
+		case "post":
+			return;
+		case "invites":
+			return;
+		case "announce":
+			return;
+		case "userinfo":
+			if (commandAndArgs.length < 2) {
+				res.status(400).send("Invalid number of arguments, expected 1, got " + (commandAndArgs.length - 1) + ".");
+				return;
+			}
+			res.status(200).send(getMemberDataByCriteria(["user_id", "username", "roles", "joined", "last_seen", "preferences", "verification", "username_history"],
+				"username",
+				commandAndArgs[1]));
+			logEvents("Command executed: " + command + "\n" + getMemberDataByCriteria(["user_id", "username", "roles", "joined", "last_seen", "preferences", "verification", "username_history"],
+				"username",
+				commandAndArgs[1]) + "\n", "adminCommands");
+			return;
+		case "help":
+			if (commandAndArgs.length === 1) {
+				res.status(200).send("Commands: ban, delete, username, logout, verify, post, invites, announce, userinfo, help\nUse help <command> to get more information about a command.");
+				return;
+			}
+			switch (commandAndArgs[1]) {
+				case "ban":
+					res.status(200).send("Syntax: ban email|ip|browser <username> <reason> [days]\nBans a user for a duration for the given reason.");
```

Comment:
I think the ban command is just going to accept a username argument, and auto-ban their email, ip, and browser from that.